### ISSUE TITLE
GnuPG Supporting PHP7+

### ISSUE THREAD

**Issue by thevalo** [maryo/php-5.5-windows-extensions#10]
Hello, where can I find, or can anybody compile resources of php_gnupg to support the latest version PHP 7.X? I'm having problems compiling it myself, or if somebody wants to help to compile then I may create a repo and publish the DLL resources.
Thanks.

**Comment by maryo** [comment]
@thevalo Hi, I doubt there is a Windows DLL available because the project contains only unix configuration file. But since I was somehow able to compile it few years ago then I had to create custom `config.w32` file to compile it but to be honest I am not sure if the extension works well. Or maybe it could be compiled on cygwin but I've never tried it.

I won't be compiling it but you can try it yourself if you're so interested.
There are some guides how to do it like the official one https://wiki.php.net/internals/windows/stepbystepbuild

You will need to download gpg4win and use it's header files and lib files. But I'm not sure if it's still compatible and if so then what version is needed. You'd probably need to experiment. Perhaps you should start compiling an extension which already has the config.w32 so you can see how it works.

Good luck :-)

**Comment by thevalo** [comment]
Hi, great to see you active on this repo again.
I actually followed some tutorials on the internet, however although converting from `.m4` to `.w32` was stated as menial, i.e, not requiring much work, I did not actually see practically such things, and so without it it can't be converted into a dynamic link library. If I'm not mistaken, there was some IFDEFs WIN32 in php_gnupg `h` files, that means it is meant to work under `win` environment too?

Anyhow, to compile it I first must have the gpg's `h` and `lib`s? Also, do you know any help about "converting" `m4` to `w32`?

Thank you.

**Comment by maryo** [comment]
> there was some IFDEFs WIN32 in php_gnupg h files, that means it is meant to work under win environment too?

I don't know. Perhaps it was meant to be compiled using Cygwin on Windows but then I am afraid that it would not be compatible with the official distribution compiled using Visual Studio environment but I am not sure. I'm also not such an expert on this topic :-).

Yes. You need the same headers (`h`s) and libraries (`so`s on linux, `lib`s on windows) which are linked in the m4 version. You will need to find them somewhere on the internet. I probably used the ones from the `gpg4win-vanilla-2.2.1.exe`

 So basically you just need to copy the lib files to proper directory and include the headers using the w32 file.

For example
https://github.com/krakjoe/apcu/blob/master/config.m4
vs
https://github.com/krakjoe/apcu/blob/master/config.w32

Both files contain `PHP_INSTALL_HEADERS`

and then build the extension as a dynamic library.

Good luck.
